source/Applications/Basic/Visualization/CaptureHDRVisNormals/CaptureHDRVisNormals.cpp

```diff
@@ -0,0 +1,123 @@
+            };
+            settings.acquisitions().emplaceBack(acquisitionSettings);
+        }
+
+        std::cout << "Capturing frame (HDR)" << std::endl;
+        const auto frame = camera.capture(settings);
+        const auto pointCloud = frame.pointCloud();
+        const auto points = pointCloud.copyData<Zivid::PointXYZColorRGBA>();
+        const auto normals = pointCloud.copyData<Zivid::NormalXYZ>();
+
+        std::cout << "Creating PCL point cloud structure" << std::endl;
+        auto cloud = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>();
+        cloud->width = pointCloud.width();
+        cloud->height = pointCloud.height();
+        cloud->is_dense = false;
+        cloud->points.resize(pointCloud.size());
+        auto cloud_normals = boost::make_shared<pcl::PointCloud<pcl::PointXYZRGBNormal>>();
+        cloud_normals->width = pointCloud.width();
+        cloud_normals->height = pointCloud.height();
+        cloud_normals->is_dense = false;
+        cloud_normals->points.resize(pointCloud.size());
+        for(size_t i = 0; i < pointCloud.size(); ++i)
+        {
+            cloud->points[i].x = points(i).point.x;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud->points[i].y = points(i).point.y;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud->points[i].z = points(i).point.z;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud->points[i].r = points(i).color.r;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud->points[i].g = points(i).color.g;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud->points[i].b = points(i).color.b;         // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].x = points(i).point.x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].y = points(i).point.y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].z = points(i).point.z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            // Scale colors for normals between 0 and 1 to create a normal map
+            cloud_normals->points[i].r =
+                0.5 * (1 - normals(i).x) * 255; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].g =
+                0.5 * (1 - normals(i).y) * 255; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].b =
+                0.5 * (1 - normals(i).z) * 255;               // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].normal_x = normals(i).x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].normal_y = normals(i).y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            cloud_normals->points[i].normal_z = normals(i).z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
```

Comment:
I think this is occupying most of the main function. How about creating a function for it, e.g. zividToPCLPointCloud(data,normals)?

See what we did here for Halcon format: https://github.com/zivid/zivid-halcon-samples/blob/master/source/cpp/Camera/Basic/CaptureViaZivid/CaptureViaZivid.cpp

Reply:
Done in f67b432119da58070a50300664e03f8315f1f695